themes/frontierline/functions.php

```diff
@@ -65,6 +66,23 @@ function frontierline_activate() {
 add_action('after_switch_theme', 'frontierline_activate');
 
 
+/**
+ * Add discover/download Firefox to the global menu.
+ */
+function frontierline_top_menu_firefox_button( $items = '', $args = array( 'theme_location' => 'global_menu' ) ) {
+  if( ((object)$args)->theme_location === 'global_menu' )  {
```

Comment:
This line is throwing an error:
`Parse error: syntax error, unexpected '->' (T_OBJECT_OPERATOR)`

From a bit of googling this may be something about calling a method on an object instead of a variable? If I change the line to simply:

`if ( $args->theme_location === 'global_menu' ) {`

It seems to work as expected, but that may not be the actual solution, I was just trying it as part of debugging. There may be some more obvious syntax error I'm just overlooking. This isn't really my area of expertise so I would defer to someone more skilled with PHP.

Reply:
Let me try to refactor it. What version of PHP do you have? I had a similar problem problem without the explicit cast.

Does the error when the function is called with the default `$args`, or when there is an actual value?

Reply:
PHP 5.6, but production is on PHP7. If this is a recent syntax it might be fine in prod. I can test more.

Reply:
Tested again with PHP7 and it works as intended. However, maybe we should still consider refactoring the function in the name of backwards compatibility. Otherwise we'd have to explicitly require a minimum of PHP7 to run this theme, and that's still a fairly recent update that not every host will be on (my personal site is still on PHP5, for example).

Reply:
Thanks. I have been developing with PHP 7. I will give it a spin with PHP 5.6 too. Is the minimum version 5.6 ok, or should I test with older versions too?

Reply:
I think 5.6 should be pretty safe as a minimum requirement. Based on stats at https://nettrack.info/php_versions.html 5.6 is clearly the most dominant.